srcs/runners_manager/vm_creation/gcloud/GcloudManager.py

```diff
@@ -120,7 +120,7 @@ def configure_instance(
         automatic_restart = True
         provisioning_model = "STANDARD"
         instance_termination_action = "DEFAULT"
-        preemptible = bool(runner.vm_type.config.get("spot", False))
+        preemptible = str(runner.vm_type.config.get("spot", "")).lower() == "true"
```

Comment:
I think this is due to how Python's `bool` method behaves, which will consider the following:
```python
>>> bool("false")
True
>>> bool("False")
True
```

Try using the [strtobool](https://docs.python.org/3/distutils/apiref.html#distutils.util.strtobool) method, see if it helps solving our issue.

```suggestion
# assuming the following import is made
from distutils.util import strtobool

        preemptible = strtobool(runner.vm_type.config.get("spot", False))
```

Reply:
yeah this works

Reply:
then let's apply it.